common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -342,6 +358,42 @@ private static int collationNameToId(String collationName) throws SparkException
       }
 
       protected abstract Collation buildCollation();
+
+      protected String catalog;
+      protected String schema;
```

Comment:
Let's make these fields part of Collation class, to avoid this unnecessary calls for setCatalog and setSchema. And to make collation object contain all the information about given collation and not get some other information from other place.

Reply:
Do you mean we remove the `Meta` class from here and let the `Collation` include all the `fields` of the `Meta`?

Reply:
I would say that is the best way to go, as that actually is the part of Collation.